lib/tasks/wikirate.rake

```diff
@@ -48,8 +87,15 @@ namespace :wikirate do
         ActiveRecord::Base.connection.execute 'drop table card_revisions'
         ActiveRecord::Base.connection.execute 'drop table users'
         ActiveRecord::Base.connection.execute 'delete from sessions'
-        ActiveRecord::Base.connection.execute "delete ca from cards ca inner join cards le ON ca.left_id = le.id where le.type_id in ('631' '651' '699' '974' '1010' '1109' '1591' '1638' '1690' '2207' '2317' '2327' '2754' '2755' '2813' '2995' '4010' '4030') and ca.created_at < '2014'" # delete all webpage++link
-        ActiveRecord::Base.connection.execute "delete from cards where type_id in ('631' '651' '699' '974' '1010' '1109' '1591' '1638' '1690' '2207' '2317' '2327' '2754' '2755' '2813' '2995' '4010' '4030') and created_at < '2014'"
+        
+        # delete companies
+        ActiveRecord::Base.connection.execute "delete from cards where type_id = '651' and id not in ( #{company_ids[0...-1]} )"
+        # delete topics
+        ActiveRecord::Base.connection.execute "delete from cards where type_id = '1010' and id not in ( #{topic_ids[0...-1]} )"
+
+        # delete all webpage++link
+        ActiveRecord::Base.connection.execute "delete ca from cards ca inner join cards le ON ca.left_id = le.id where le.type_id in ('631','651','699','974','1010','1109','1591','1638','1690','2207','2317','2327','2754','2755','2813','2995','4010','4030') and ca.id not in  ( #{card_to_be_kept[0...-1]} )" 
+        ActiveRecord::Base.connection.execute "delete from cards where type_id in ('631','699','974','1109','1591','1638','1690','2207','2317','2327','2754','2755','2813','2995','4010','4030') and id not in ( #{card_to_be_kept[0...-1]} )"
         puts "clean database"
```

Comment:
I know you didn't just add this, but what is all of this stuff?  It would be much more legible if we didn't use id's in here.

For example, you could do:
````
%w{ Company Topic Phrase Whatever }.map { |typename] Card.fetch_id(typename) }
````

Similarly you could do 
````
"....where type_id = '#{Card::WikirateCompanyID}'...."
````

As it stands, I (or future you) could only figure out the code by looking at the data...